src/diagram/models/variable.ts

```diff
@@ -185,7 +185,9 @@ export const Variable = types.model("Variable", {
     self.inputB = newInputB;
   },
   setValue(newValue?: number) {
-    self.value = newValue;
+    // NaN and Infinity are not valid JSON, we can probably fix MST to handle this correctly
+    // but for now we just convert it to undefined to be safe.
+    self.value = newValue !== undefined && isFinite(newValue) ? newValue : undefined;
```

Comment:
I would use `!= null` for breadth of coverage although the distinction may not matter in this instance.

Reply:
Good point.